build_node/build_node_builder.py

```diff
@@ -117,17 +118,19 @@ def run(self):
                         build_artifacts = self.__upload_artifacts(
                             task, artifacts_dir, task_log_file)
                         if excluded_exception is not None:
-                            self.__report_excluded_task(
-                                task, build_artifacts)
+                            while not self.__task_completed:
```

Comment:
It will stuck if a build node will not be available. Add some counter for both loops